### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Ask VA Integration - Ask VA

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#71169]
## VFS actions
- [ ] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] Close the ticket when all feedback has been addressed.

## Thoughts/questions

- This is a huge product, and you're doing a ton of things exactly right. Thank you for the thorough artifacts and the very helpful discussion on Friday!
- I very much appreciate the annotations! They make my job significantly easier.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Chapters 1-2
- **Consider:** On the [Ask VA landing page](https://www.sketch.com/s/8cf9349c-9860-4f4e-9ae2-ac819b1ab030/a/uuid/B23FD2DA-5B34-48CF-8734-A9FE3E8C4963), I think there's a potential cognitive load issue with the "Signing in is required..." alert heading and the "Sign in for the best experience" heading. They're describing distinct use cases, but a user scanning the page quickly might get confused (Do I have to sign in, or is it just better if I do? Why are they telling me to sign in twice?) and may need an extra pass to parse it. I'm wondering if there's a content solution that communicates this in a less duplicative way.
- **Must:** Be careful about material honesty, especially around buttons and links. On the [landing page](https://www.sketch.com/s/8cf9349c-9860-4f4e-9ae2-ac819b1ab030/a/uuid/B23FD2DA-5B34-48CF-8734-A9FE3E8C4963), there's a button and a link that both say "Sign in or create an account." If they do the same thing, they should be the same element (if navigational then link; if action like opening a modal then button). If they do different things, they should probably have different text.
- **Must:** I absolutely LOVE the unique H3s on each page, but as presented here I think there's going to be an assistive tech issue for inputs that are labeled by the `label` element. A `label` can't also be a heading without some weird screen reader behavior --- and I don't think there's a way to do it with out-of-box design system components either. For those fields, you'll likely need to have a little but of redundant text, eg. on [step 1](https://www.sketch.com/s/8cf9349c-9860-4f4e-9ae2-ac819b1ab030/a/uuid/C64074D8-5016-4B7C-B4D5-4EBB1A22E24D) you might have an H3 of "Which category best describes your question?" and a `label` of "Select a category."
   
   For inputs that are collected in a `fieldset` and labeled with a `legend`, this is less of an issue. No assistive tech issues for a `legend` that contains an `h3`.
- **Must:** When setting focus moving from screen to screen, send focus to the unique H3 with `tabindex="-1"`. That's going to ensure the H3 is always the first thing announced by screen readers when landing on the page, and will give a consistent experience rather than getting input semantics immediately after landing on the screen.
- **Consider:** This is more in the content territory but may lead to some extra cognitive load, but I was a little thrown off on the ["Your question" screen](https://www.sketch.com/s/8cf9349c-9860-4f4e-9ae2-ac819b1ab030/a/uuid/D5EAAF51-5C84-4AD3-BE2C-FF7A66B7F9AA) with the phrasing of the options in the "Tell us the reason" radio buttons. Everything else on the page says "Your question" and "What's your question," then there's this follow-up question that suggests you could be providing something that's not actually a question. It seems like a strange way to present it and might lead to confusion.

### Chapter 3
- **Must:** Same note as in Chapters 1-2 about unique H3s.
- **Must:** Same note as in Chapters 1-2 about focus going to the H3.

### Chapter 4
- No feedback!

### Dashboard
- **Must:** Echoing the note in Chapters 1-2, be careful about material honesty around links and buttons. On the [dashboard](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/a/uuid/968CEC3D-01D6-43E6-A398-F96246A2BE09), my best guess is that "Create new question" is navigational, taking the user to the first chapter of the big workflow. If so, that should be a link rather than a button.
- **Should:** On the [card display for past questions](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/a/uuid/AE9CFBB7-B6B2-48A9-838D-1B3E75141F9B), the dropdowns for filtering (last updated and status) should have a separate button to initiate the filter action, rather than filtering happening just on changing the selection for each dropdown. That might impact the card/list toggle (see below for more on those buttons).
- **Must:** On the [card display for past questions](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/a/uuid/AE9CFBB7-B6B2-48A9-838D-1B3E75141F9B), I would recommend some kind of line of text between the filtering options and the cards themselves that gives a summary of how the questions are currently sorted --- eg. "Showing 4 cards, newest to oldest with all statuses" or similar. Then, when a user filters, that text updates and receives keyboard focus. This would ensure that the status message is announced to screen reader users, so they know that a change occurred and what's now being displayed (since they can't rely on visual cues). Without something like this, screen reader users may not get any feedback to know that something happened. I'm imagining something similar to the message that receives focus on the [VA.gov search tool](https://www.va.gov/search/?query=disability&t=false) when you enter a new term.
- **Must:** The buttons for switching from card to list view don't have any visible text associated with them, which will likely be an issue for sighted voice command software users. When a user sees a link or button that they want to click, they'll typically speak a command like "Click [button text]." But if there's just an icon and no text, they're left trying to guess the magic words for that button. Adding visible text probably changes how you'll want these to be positioned, but the text helps a ton for those users.
- **Must:** The "Check details" links on each item (card or list view) will need a unique accessible name. Screen reader users often scan the page by having their screen reader announce a list of all of the links, then jumping straight to the link that sounds relevant to what they want. If they hear a bunch of "Check details" links in a row, that doesn't provide much context and doesn't help them decide where to go. The best option here is visible text for all users, eg. link text that says "Check details for your August 20, 2023 question" or something similar. Since that adds a lot of visual clutter, using an `aria-label` for screen reader users would be sufficient, eg. `<a href="..." aria-label="Check details for your August 20, 2023 question">Check details</a>`.
- **Consider:** As discussed in the meeting, I'm okay with the tabs for business/personal questions. It's a limited set of users who will encounter it, and the design system's tab component is okay-but-not-great, which is fine for this. But I would love to hear about any other design explorations you do around this!
- **Should:** I don't have any specific concerns about the [question details page](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/a/uuid/3752EB05-ABB2-4BF4-B394-A60E7458700B), but I would encourage you to review the work being done on MyHealtheVet's Secure Messaging tool and see if there's any opportunity to synchronize with them. The message exchange here is not exactly the same as what Secure Messaging is doing, but it's similar enough that a similar layout or a similar organization of sections may help reduce cognitive load.

### Search by reference number
- **Must:** When coded, make sure that when a user searches for reference number keyboard focus then moves to the next relevant content (either the "nope nothing found" message or the status line when something is found). Similar reasoning as with card filtering above --- screen reader users need to hear something after they perform the action to know that something has happened, and we want to give them whatever is most useful and relevant.

**Comment by tygindraux** [comment]
Hi Brian (@briandeconinck)! Thank you so much for this incredible detailed feedback. It's a pleasure learning from you.

Everything you've listed makes sense, so we'll take this away and comment again once we've made updates.

**Comment by t-michaud** [comment]
@briandeconinck - I have a question about:
Should: On the [card display for past questions](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/a/uuid/AE9CFBB7-B6B2-48A9-838D-1B3E75141F9B), the dropdowns for filtering (last updated and status) should have a separate button to initiate the filter action, rather than filtering happening just on changing the selection for each dropdown. That might impact the card/list toggle (see below for more on those buttons).

- Could you point me to examples of this pattern you're suggesting?
   - In most filters (or sorts) the user doesn't click on a button to "apply" ... I certainly understand adding in content that comes into focus (which I've looked at adding). I've also added in an alternative (in the box to the far right) that has a button to "apply" the filter.

Could both sorting and filtering interactions be solved through moving focus to the "results" (after the submitter has made their selection)? I've put designs together in [Sketch Ask VA Dashboard](https://www.sketch.com/s/bf31f900-9da5-4d65-8a55-b44949e3cdfb/p/B5883873-3480-4BB9-A41D-EA229F7B6157/canvas#Version) (and added a screenshot below)

**Sorting**
Questions are showing the oldest updated questions first.
Questions are showing the most recently updated questions first.

**Filtering**
There are [number] questions matching the status of [status filter name]
You are viewing all questions in your dashboard. [default]
There are **4 questions** matching the status of **In progress**.
There are **no questions** matching the status of **Reopened**.  Try a different filter.

 
![Components - Sort and Filter](https://github.com/department-of-veterans-affairs/va.gov-team/assets/135402634/26cb24da-c01f-44ed-a0d4-4d10639a7569)
